addons/mail/models/mail_address_mixin.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, tools, _
+from odoo.exceptions import UserError
+
+
+class MailAddressMixin(models.AbstractModel):
+    """ This Mixin can be used to store a normalized version of the primary email address of the parent model.
+        The email_normalized field can than be used on that model to search quickly on emails (by simple comparison
+        and not using time consuming regex anymore).
+        USAGE : the field '_primary_email' must be overridden in the model that inherit the mixin
+        and must contain the email field of the model.
+        """
+    _name = 'mail.address.mixin'
+    _description = 'Email address mixin'
+    _primary_email = ['email']
+
+    email_normalized = fields.Char(string='Normalized email address', compute="_compute_email_normalized", invisible=True,
+                                  compute_sudo=True, store=True, help="""This field is used to search on email address,
+                                  as the primary email field can contain more than strictly an email address.""")
+
+    @api.depends(lambda self: self._primary_email)
+    def _compute_email_normalized(self):
+        self._assert_primary_email()
+        [email_field] = self._primary_email
+        for record in self:
+            record.email_normalized = self._normalize_email(record[email_field])
+
+    def _normalize_email(self, email):
```

Comment:
Shouldn't we put this method in tools/mails.py ?

Reply:
done in one of the next commits